crates/catalog/sql/src/catalog.rs

```diff
@@ -0,0 +1,517 @@
+        let catalog_name = self.name.clone();
+        let namespace = identifier.namespace().encode_in_url();
+        let name = identifier.name().to_string();
+        let rows = sqlx::query(
+            &("select ".to_string()
+                + TABLE_NAMESPACE
+                + ", "
+                + TABLE_NAME
+                + ", "
+                + METADATA_LOCATION_PROP
+                + ", "
+                + PREVIOUS_METADATA_LOCATION_PROP
+                + " from "
+                + CATALOG_TABLE_VIEW_NAME
+                + " where "
+                + CATALOG_NAME
+                + " = ? and "
+                + TABLE_NAMESPACE
+                + " = ? and "
+                + TABLE_NAME
+                + " = ?;"),
+        )
+        .bind(&catalog_name)
+        .bind(&namespace)
+        .bind(&name)
+        .fetch_all(&self.connection)
+        .await
+        .map_err(from_sqlx_error)?;
+        let mut iter = rows.iter().map(query_map);
+
+        Ok(iter.next().is_some())
+    }
+
+    async fn drop_table(&self, _identifier: &TableIdent) -> Result<()> {
+        todo!()
+    }
+
+    async fn load_table(&self, identifier: &TableIdent) -> Result<Table> {
+        let metadata_location = {
+            let catalog_name = self.name.clone();
+            let namespace = identifier.namespace().encode_in_url();
+            let name = identifier.name().to_string();
+            let row = sqlx::query(
+                &("select ".to_string()
+                    + TABLE_NAMESPACE
+                    + ", "
+                    + TABLE_NAME
+                    + ", "
+                    + METADATA_LOCATION_PROP
+                    + ", "
+                    + PREVIOUS_METADATA_LOCATION_PROP
+                    + " from "
+                    + CATALOG_TABLE_VIEW_NAME
+                    + " where "
+                    + CATALOG_NAME
+                    + " = ? and "
+                    + TABLE_NAMESPACE
+                    + " = ? and "
+                    + TABLE_NAME
+                    + " = ?;"),
+            )
+            .bind(&catalog_name)
+            .bind(&namespace)
+            .bind(&name)
+            .fetch_one(&self.connection)
+            .await
+            .map_err(from_sqlx_error)?;
+            let row = query_map(&row).map_err(from_sqlx_error)?;
+
+            row.metadata_location
+        };
+        let file = self.storage.new_input(&metadata_location)?;
+
+        let mut json = String::new();
+        file.reader().await?.read_to_string(&mut json).await?;
+
+        let metadata: TableMetadata = serde_json::from_str(&json)?;
+
+        self.cache
+            .insert(identifier.clone(), (metadata_location, metadata.clone()));
+
+        let table = Table::builder()
+            .file_io(self.storage.clone())
+            .identifier(identifier.clone())
+            .metadata(metadata)
+            .build();
+
+        Ok(table)
+    }
+
+    async fn create_table(
```

Comment:
There are some things missing here:
1. We should first check namespace exists
2. The location is optional, it should use warehouse's subdir as location

I would suggest to refer to [python's implementation](https://github.com/apache/iceberg-python/blob/a892309936effa7ec575195ad3be70193e82d704/pyiceberg/catalog/sql.py#L146),